contracts/TwabRewards.sol

```diff
@@ -0,0 +1,394 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@pooltogether/v4-core/contracts/interfaces/ITicket.sol";
+
+import "./interfaces/ITwabRewards.sol";
+
+/**
+ * @title PoolTogether V4 TwabRewards
+ * @author PoolTogether Inc Team
+ * @notice Contract to distribute rewards to depositors in a pool.
+ */
+contract TwabRewards is ITwabRewards {
+    using SafeERC20 for IERC20;
+
+    /* ============ Global Variables ============ */
+
+    /// @notice Settings of each promotion.
+    mapping(uint256 => Promotion) internal _promotions;
+
+    /// @notice Latest promotion id.
+    uint256 internal _latestPromotionId;
+
+    /// @notice Keeps track of claimed rewards per user.
+    /// @dev _claimedEpochs[promotionId][user] => claimedEpochs
+    /// @dev We pack epochs claimed by a user into a uint256. So we can't store more than 256 epochs.
+    mapping(uint256 => mapping(address => uint256)) internal _claimedEpochs;
+
+    /* ============ Events ============ */
+
+    /**
+        @notice Emmited when a promotion is created.
+        @param id Id of the newly created promotion
+    */
+    event PromotionCreated(uint256 id);
+
+    /**
+        @notice Emmited when a promotion is extended.
+        @param token Address of the token used in the promotion
+        @param amount Amount of tokens transferred to the rewards contract
+    */
+    event PromotionCancelled(IERC20 token, uint256 amount);
+
+    /**
+        @notice Emmited when a promotion is extended.
+        @param token Address of the token used in the promotion
+        @param amount Amount of tokens transferred to the recipient address
+        @param numberOfEpochs New number of epochs after extending the promotion
+    */
+    event PromotionExtended(IERC20 token, uint256 amount, uint256 numberOfEpochs);
+
+    /**
+        @notice Emmited when rewards have been claimed.
+        @param token Address of the token used in the promotion
+        @param amount Amount of tokens transferred to the recipient address
+    */
+    event RewardsClaimed(IERC20 token, uint256 amount);
```

Comment:
should include the promotion id + epoch id  and remove the token address

Reply:
Done: https://github.com/pooltogether/v4-periphery/pull/8/commits/0e39681caa48d861f4840ac01450983bdf95caaa